packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -198,42 +213,50 @@ fn public_storage_fns(
     };
 
     match &value {
-        Value::Status => value_status_fns(variant, &param_list, &storage_method, key),
+        Value::Status => value_status_fns(
+            &param_list,
+            &storage_method,
+            key,
+            &ttl_fn,
```

Comment:
Looks like the `&key` is implicitly captured in `ttl_method` implementation which is not good. key should be passed in to `ttl_method` to get the `ttl_fn`